meson.build

```diff
@@ -0,0 +1,797 @@
+    common_ldflags   += ['--coverage']
+    common_cflags    += coverage_cflags
+    libmcount_cflags += coverage_cflags
+    test_cflags      += coverage_cflags
+endif
+
+################################
+# -Dwith-capstone build option #
+################################
+
+libcapstone_dep = dependency('capstone', required: get_option('with-capstone'))
+libcapstone_is_enabled = libcapstone_dep.found()
+if libcapstone_is_enabled
+    common_cflags += '-DHAVE_LIBCAPSTONE'
+endif
+
+##############################
+# -Dwith-libdw  build option #
+# -Dwith-libelf build option #
+##############################
+
+libelf_dep = dependency('libelf', required: get_option('with-libelf'))
+libelf_is_enabled = libelf_dep.found()
+if libelf_is_enabled
+    common_cflags += '-DHAVE_LIBELF'
+
+    libdw_dep = dependency('libdw', required: get_option('with-libdw'))
+    libdw_is_enabled = libdw_dep.found()
+    if libdw_is_enabled
+        common_cflags += '-DHAVE_LIBDW'
+    endif
+else
+    libdw_dep = []
+    libdw_is_enabled = false
+endif
+
+#################################
+# -Dwith-libluajit build option #
+#################################
+
+libluajit_dep = dependency('luajit', required: get_option('with-libluajit'))
+libluajit_is_enabled = libluajit_dep.found()
+if libluajit_is_enabled
+    common_cflags += '-DHAVE_LIBLUAJIT'
+endif
+
+###################################
+# -Dwith-libncursesw build option #
+###################################
+
+libncursesw_dep = dependency('ncursesw', required: get_option('with-libncursesw'))
+libncursesw_is_enabled = libncursesw_dep.found()
+if libncursesw_is_enabled
+    common_cflags += '-DHAVE_LIBNCURSES'
+endif
+
+#################################
+# -Dwith-libpython build option #
+#################################
+
+python_installation = python_module.find_installation('python', required: get_option('with-libpython'))
+libpython_dep = []
+libpython_is_enabled = python_installation.found()
+if libpython_is_enabled
+    libpython_dep = python_installation.dependency()
+    python_version = python_installation.language_version()
+    common_cflags += '-DLIBPYTHON_VERSION=@0@'.format(python_version)
+    common_cflags += '-DHAVE_LIBPYTHON' + python_version[0]
+endif
+
+##################################
+# -Dwith-libunwind build option #
+##################################
+
+libunwind_dep = dependency('libunwind', required: get_option('with-libunwind'))
+libunwind_is_enabled = get_option('debug') and libunwind_dep.found()
+if libunwind_is_enabled
+    common_cflags += '-DHAVE_LIBUNWIND'
+endif
+
+#################################
+# -Dwith-libstdcxx build option #
+#################################
+
+cxa_demangle_is_enabled = false
+cxa_demangle_dep = []
+have_cxa_demangle_code = run_command('cat', files('check-deps/__cxa_demangle.c')).stdout().strip()
+have_cxa_demangle_compiles = cc.compiles(have_cxa_demangle_code, name: 'cxa_demangle check', args: ['-lstdc++'])
```

Comment:
`cc.compiles(files('check-deps/__cxa_demangle.c'))` should work, without needing to first cat it into a meson variable.